runtime/obj.c

```diff
@@ -114,6 +119,12 @@ CAMLprim value caml_obj_block(value tag, value size)
     Field (res, sz - 1) = 0;
   }
 
+  if (tg == Closure_tag) {
+    /* Closinfo_val is the seconnd field, so we need size at least 2 */
```

Comment:
Typo: "seconnd".

Cosmetic: `switch (tg) { case Closure_tag:  ... case String_tag: ... }` could look good. Maybe.

Reply:
I did the switch (haha) in a separate commit, let me know if you like it.